### PR TITLE
Add some assertions

### PR DESCRIPTION
**Pull request by kssenii**
### Changelog category (leave one):
- Not for changelog (changelog entry is not required)

### CODE REVIEW

**Comment by robot-clickhouse** [comment]
This is an automated comment for commit 79c14c89ee39bea6017e41e375f9b6a59e3e7899 with description of existing statuses. It's updated for the latest CI running
The full report is available [here](https://s3.amazonaws.com/clickhouse-test-reports/50470/79c14c89ee39bea6017e41e375f9b6a59e3e7899/ci_running.html)
The overall status of the commit is 🟢 success

<table><thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody><tr><td>Style Check</td><td>Runs a set of checks to keep the code style clean. If some of tests failed, see the related log from the report</td><td>🟢 success</td></tr>
</table>

**Comment by kssenii** [comment]
> Stateless tests (msan) [3/6] — fail: 1

`00534_functions_bad_arguments12`

> AST fuzzer (asan) — Received signal 11

https://github.com/ClickHouse/ClickHouse/issues/50451

**Comment by kssenii** [comment]
> Integration tests (release) [4/4] — fail: 2

`test_ssl_cert_authentication` - not related to changes

> Stateless tests (ubsan) [2/2] — fail: 2

`01676_clickhouse_client_autocomplete`
`00992_system_parts_race_condition_zookeeper_long` - not related to changes


>  ClickHouse build check — 21/22 artifact groups are OK 

I do not see the reason why it failed... but it passed in previous commit, the last commit changed only 1 line of code, which does not seem to be able to break the build

**Review by nickitat — APPROVED** [review]

File: src/Interpreters/Cache/FileSegment.cpp [hunk 1]
```diff
@@ -376,9 +378,10 @@ void FileSegment::write(const char * from, size_t size, size_t offset)
         int code = e.getErrno();
         if (code == /* No space left on device */28 || code == /* Quota exceeded */122)
         {
-            const auto file_size = fs::file_size(getPathInLocalCache());
+            const auto file_size = fs::file_size(file_segment_path);
             chassert(downloaded_size <= file_size);
             chassert(reserved_size >= file_size);
+            chassert(file_size <= range().right + 1);
```
[the remark this answers is not in the document]
Comment (kssenii) [reply src/Interpreters/Cache/FileSegment.cpp:384]: A case: we had `write n bytes` in `FileSegment::write`, have written `k < n`, but then had an exception about `not space left on device`, so we need to account this successfully written `k`. So this is done in lines `385-386`
And I added this assertion `chassert(file_size <= range().right + 1);` it is just a paranoid check, that file size is sane and does not exceed file segment size.

[the remark this answers is not in the document]
Comment (kssenii) [reply src/Interpreters/Cache/FileSegment.cpp:384]: > that file size is sane and does not exceed file segment size.

ops, we need here 
```suggestion
            chassert(file_size <= range().size);
```
...

[the remark this answers is not in the document]
Comment (nickitat) [reply src/Interpreters/Cache/FileSegment.cpp:384]: ok

**Pull request merged**